### PR TITLE
organisations: access records with specific URL

### LINKED ISSUE #216: Possibility to access records for an organisation with a specific URL
All organisations can be reached.

### PR DESCRIPTION
**Pull request by sebastiendeleze**
* Adds a property `isShared` in organisation to allow access to records by a specific URL.
* Adds a property `isDedicated` in organisation to allow organisation to have a specific theme and logo.
* Adds a specific route to document details instead of built-in invenio view.
* Adds a method to get current view code in templates.
* Renames `ir` query parameter to `view` for consistency.
* Stores current organisation record in Flask globals.
* Adds a header containing current organisation information in templates.
* Checks if organisation is flagged as dedicated for including custom logo and styles.
* Closes #216.

Co-Authored-by: Sébastien Délèze <sebastien.deleze@rero.ch>

### Shared repository
1. Try to access /organisation/unisi --> Error is thrown.
2. Login as superuser.
3. Edit unisi organisation and check option "Is shared".
4. Try to access /organisation/unisi --> Is accessible.

### Dedicated.
1. Login as superuser.
2. Edit unisi organisation and check option "Is shared" and "Is dedicated".
3. Go to /organisation/unisi --> Page should have a custom theme.

### CODE REVIEW

File: sonar/modules/documents/views.py [hunk 1]
```diff
@@ -43,33 +45,50 @@
...
+def search():
+    """Search results page."""
     return render_template('sonar/search.html')
 
 
-def detail(pid, record, template=None, **kwargs):
-    """Search details."""
-    g.ir = kwargs.get('ir')
-    return render_template('documents/record.html', pid=pid, record=record)
+@blueprint.route('/documents/<pid_value>')
```
Comment (jma) [inline sonar/modules/documents/views.py:81]: Seems a conflict with https://github.com/rero/sonar/blob/5af14d6daab1edb806264e8bae8708b58d003d00/sonar/config.py#L246

File: sonar/modules/documents/views.py [hunk 2]
```diff
@@ -389,13 +408,16 @@ def get_preferred_languages(force_language=None):
     return list(dict.fromkeys(preferred_languages))
 
 
-def get_current_organisation():
+def get_current_organisation_code():
```
Comment (jma) [inline sonar/modules/documents/views.py:411]: this can be probably a decorator

File: sonar/theme/templates/sonar/partial/navbar.html [hunk 3]
```diff
@@ -15,86 +15,100 @@
...
+        <div class="col text-right">
+          <a class="text-light" href="{{url_for('documents.index', view=config.SONAR_APP_DEFAULT_ORGANISATION)}}">
+            {{_('Back to SONAR')}}
+          </a>
+        </div>
+      </div>
+    </div>
+  </div>
+{% endif %}
+<nav class="navbar navbar-expand-lg navbar-dark bg-{{ 'header' if page == 'home' else 'ir' }}">
```
Comment (jma) [inline sonar/theme/templates/sonar/partial/navbar.html:32]: still ir?

File: sonar/modules/documents/views.py (hunk 1 above)
Comment (sebastiendeleze) [reply sonar/modules/documents/views.py:81]: I expressly removed it in this commit: https://github.com/rero/sonar/pull/234/files#diff-8cdd8777bd2125b8f78dfb443c855eedL246-L250.

**Pull request merged**